### ISSUE TITLE
Special case to preserve/ban `implicit-optional` when baselined

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#152]
Turning off implicit-optional and baselining will introduce a bunch of false errors
```py
def foo(value: str = None):  # baselined "assignment" error
  bar: str = value # no error because typeof value is `str`
```
We should implement a special case to either: 
- straight up ban this code even when baselined
- preserve the implicit optional
